apps/prairielearn/src/migrations/20231031163354_client_fingerprints__create.sql

```diff
@@ -0,0 +1,37 @@
+CREATE TABLE
+  client_fingerprints (
+    id BIGSERIAL PRIMARY KEY,
+    user_id BIGINT REFERENCES users (user_id),
+    user_session_id BIGINT REFERENCES user_sessions (id),
+    ip_address INET NOT NULL,
+    user_agent VARCHAR(255),
+    accept_language VARCHAR(255),
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now()
+  );
+
+-- add client_fingerprint_id to user_sessions
+ALTER TABLE user_sessions
+ADD COLUMN client_fingerprint_id BIGINT REFERENCES client_fingerprints (id) ON UPDATE CASCADE ON DELETE CASCADE;
+
+-- add client_fingerprint_id to page_view_logs
+ALTER TABLE page_view_logs
+ADD COLUMN client_fingerprint_id BIGINT REFERENCES client_fingerprints (id) ON UPDATE CASCADE ON DELETE CASCADE;
+
+-- add client_fingerprint_id to submissions
+ALTER TABLE submissions
+ADD COLUMN client_fingerprint_id BIGINT REFERENCES client_fingerprints (id) ON UPDATE CASCADE ON DELETE CASCADE;
```

Comment:
I am not opposed but I am also not sure how this would be implemented. Would this track for each user that created a new variant?

Reply:
Correct, yeah. This would catch the case where one fingerprint creates it and another answers it. I suppose the variant view logs would catch this as well, though it might be nice to add this for completeness?

Reply:
I think we should add this but I think we should hold off and do it with a second PR: https://github.com/PrairieLearn/PrairieLearn/pull/8798#discussion_r1408628903

See https://github.com/PrairieLearn/PrairieLearn/issues/8924

Reply:
Ok, I'm happy to wait to do this.